pkg/api/v1/types.go

```diff
@@ -2110,6 +2110,8 @@ type PodSpec struct {
 	// If specified, the pod's scheduling constraints
 	// +optional
 	Affinity *Affinity `json:"affinity,omitempty" protobuf:"bytes,18,opt,name=affinity"`
+	// SchedulingMismatchedPredicateResults Summary of false podFitsOnNode predicates for investigating unscheduled pods.
```

Comment:
Not really, NodeSelector, Annotations, and Labels had special exemptions.  Only pure maps are allowed, and I have no reason to believe that we wouldn't add additional information to these results (what you're describing is very close to a condition).  So array of structs is the appropriate serialization.

See https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md